### ISSUE TITLE
drop callback: getting the resource ID

### ISSUE THREAD

**Issue by dingzj** [fullcalendar/fullcalendar-scheduler#27]
struggling on this, and saw your example here: http://fullcalendar.io/js/fullcalendar-scheduler-1.0.0/demos/external-dragging.html

how to get the resource id in the drop callback? So far it only get `date` from it. How do I know which resourceId it will get?

`function( date, jsEvent, ui ) { }`

**Comment by outofbounder** [comment]
Check your syntax on <code>event.resourceId</code>  Note the lowercase 'd' (I was stuck on this!).

var blockedEvent = {  
                id: event.id+'booked',
                title: '-unavailable-',
                start: moment(event.start).subtract(30, 'minutes'),
                end: moment(event.end).add(30, 'minutes'),
                allDay: false,
                resourceId: linkedRoomId,
                backgroundColor: 'red',
                editable: false,

                //rendering: 'background'  // BUG
            };

**Comment by dingzj** [comment]
@outofbounder  thanks for the info.

Actually, my question is different. For external event dropping to the calendar, you have to use http://fullcalendar.io/docs/dropping/drop/  `drop(callback)` , not the `eventDrop callback `.

I had found my ways to do it; the `drop` callback has first parameter as `date`, but can be changed to `dropLocation` which will contains `date, resourceId, allDay`.

`allDay` also need to be hacked, to properly have a true or false value.

I didn't change the sourcecode coffee file, but change in the generated js files. (Otherwise, I will show my code here.)

**Comment by outofbounder** [comment]
No problem, I've been working around a few things myself.  Have you tried something like this?
<code>drop: function(date, allDay, ev, ui, res) { // this function is called when something is dropped</code>

then using <code>res.id</code> https://github.com/bkaluza/fullcalendar/blob/master/demos/external-dragging-resource.html

**Comment by dingzj** [comment]
nah. that repository is 3 years old. definitely not working with recent full-calendar.

**Comment by trbngr** [comment]
```javascript
eventReceive: function (event, view) {
  var resourceId = event. resourceId;
  ...
}
```

**Comment by arshaw** [comment]
looks like the solution has been discovered

**Comment by 50bbx** [comment]
@arshaw This is not closed. The `eventReceive` is a **different** callback. The `drop` callback is triggered when a valid jQuery object is dropped on the calendar/scheduler, NOT when an `Event Object` is dropped on the callendar. In the scheduler, the `select` callback has `startDate, endDate, jsEvent, view, resource`. The drop callback should have the `resource`. otherwise when I drop a jQuery object I cannot convert it into a valid fullcalendar `Event Object`.

@dingzj How did you find the `dropLocation` callback? Doesn't seem to work for me.

**Comment by arshaw** [comment]
ah, now i understand. i assumed this was for `eventReceive` when it was actually an issue with `drop`.

**Comment by dingzj** [comment]
@50bbx 

Here is my changes compare to the 2.4.0 version. The key part is last difference which use `dropLocation`, not `dropLocation.start` in the `triggerExternalDrop` function. Then the `drop` callback will be like `drop : function dropEvent( dropLocation, jsEvent, ui, resource ) ` , you can easily get resource id by `dropLocation.resourceId` .

![image](https://cloud.githubusercontent.com/assets/865551/11960664/bee5a5e8-a8a1-11e5-940d-d74b9af19a52.png)

**Comment by arshaw** [comment]
this has been released in scheduler v1.2.0
http://fullcalendar.io/docs/dropping/drop/
